### PR TITLE
PT-717: Top navigation implementation

### PR DESCRIPTION
**Pull request by melniiv**
## Description :sparkles:
-replaced current header with hds
-removed deprecated components
-replaced unused reducers with default "empty" one (otherwise requires more refactoring to remove all the references)

## Issues :bug:
-translations and urls has to be fixed
## Additional notes :spiral_notepad:

### CODE REVIEW

File: src/domain/app/header/Header.tsx [hunk 1]
```diff
@@ -1,26 +1,125 @@
...
+        <Navigation.LanguageSelector
+          buttonAriaLabel={t('header:changeLanguage')}
+          className={styles.languageSelector}
+          label={t(`common:languages:${locale}`)}
+          icon={<IconGlobe />}
+        >
+          {languageOptions.map((option) => (
+            <Navigation.Item
+              key={option.value}
+              href="#"
```
Comment (Sartonon) [inline src/domain/app/header/Header.tsx:114]: Enter key not working  in these items, maybe all of them need the specific url

File: src/utils/updateLocaleParam.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export const updateLocaleParam = (
+  url: string,
+  currentLocale: string,
+  value: string
+): string => {
+  const post = url.endsWith(`/${currentLocale}`) ? '' : '/';
+  return url.replace(`/${currentLocale}${post}`, `/${value}${post}`);
+};
```
Comment (Sartonon) [inline src/utils/updateLocaleParam.ts:8]: Maybe i18n.changeLanguage can be used like previously.

File: src/domain/app/header/Header.tsx (hunk 1 above)
Comment (melniiv) [reply src/domain/app/header/Header.tsx:111]: actually its a bug of hds component i think. its changing the language, but not closing the menu even on click.

File: src/utils/updateLocaleParam.ts (hunk 2 above)
Comment (melniiv) [reply src/utils/updateLocaleParam.ts]: fixed

File: src/domain/app/header/Header.tsx (hunk 1 above)
Comment (melniiv) [reply src/domain/app/header/Header.tsx:112]: can you retest? i think with updated hds it works now

**Review by nikomakela — COMMENTED** [review]
The changes looks good to me, but since I'm not familiar with HDS guidelines and stuff, I would not like to approve this. I don't have knowledge to give feedback.

File: public/static/locales/sv/header.json [hunk 3]
```diff
@@ -7,5 +7,13 @@
     "en": "In English",
     "fi": "Suomeksi",
     "sv": "På svenska"
+  },
+  "menuToggleAriaLabel": "Menyn",
```
Comment (Sartonon) [inline public/static/locales/sv/header.json:11]: typo?

Comment (Sartonon) [inline public/static/locales/sv/header.json:11]: typo?

**Review by Sartonon — APPROVED** [review]
Looks good. Added coupke comments but it is approved now

File: public/static/locales/sv/header.json (hunk 3 above)
Comment (melniiv) [reply public/static/locales/sv/header.json:11]: nope, its swedish:)

File: src/domain/app/header/Header.tsx [hunk 4]
```diff
@@ -1,26 +1,123 @@
...
+  const [menuOpen, setMenuOpen] = React.useState(false);
+  const toggleMenu = () => setMenuOpen(!menuOpen);
+  const closeMenu = () => setMenuOpen(false);
+
+  const languageOptions: OptionType[] = React.useMemo(() => {
+    return Object.values(SUPPORTED_LANGUAGES).map((language) => ({
+      label: t(`header:languages:${language}`),
+      value: language,
+    }));
+  }, [t]);
```
[the remark this answers is not in the document]
Comment (melniiv) [reply src/domain/app/header/Header.tsx:27]: ok, i ll refactor

File: src/domain/app/header/Header.tsx [hunk 5]
```diff
@@ -1,26 +1,123 @@
...
+
+  const changeLanguage = (newLanguage: OptionType) => (
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>
```
[the remark this answers is not in the document]
Comment (melniiv) [reply src/domain/app/header/Header.tsx:30]: yes

**Pull request merged**